src/core/ast.ts

```diff
@@ -0,0 +1,24 @@
+import { Token } from './token';
+
```

Comment:
- [ ] rename this file as a typedef file

Reply:
I'd rather avoid changing file extensions based on the content. I also can't really see any benefit from having some files with `.d.ts` extension.

Currently there happens to be also a function inside here, so it can't become a `.d.ts` file.